cloud_storage/cloud_storage/overrides/file.py

```diff
@@ -52,12 +53,14 @@ def on_trash(self) -> None:
 				_("This file is attached to a submitted document and cannot be deleted"),
 				frappe.PermissionError,
 			)
-		if len(self.file_association) > 1:
-			frappe.throw(
-				_("This file is attached to multiple documents and cannot be deleted"),
-				frappe.PermissionError,
-			)
-		super().on_trash()
+		if self.is_home_folder or self.is_attachments_folder:
+			frappe.throw(_("Cannot delete Home and Attachments folders"))
+		if len(self.file_association) > 0:
+			return
+		self.validate_empty_folder()
+		self._delete_file_on_disk()
+		if not self.is_folder and len(self.file_association) > 0:
```

Comment:
The `len` check will never be true here, right? Since you're returning on line 58-59 above on that condition.

Reply:
I think its true in the rename cadence. Showing a message that a file that's still attached has been deleted is confusing.

Reply:
Should we remove the if block entirely, or add a comment here?

Reply:
Comment. Without some fancy AST tools to show how reachable that code is, I'd rather say what it's intended for even if it doesn't do anything.